pkg/model/util_test.go

```diff
@@ -154,3 +154,33 @@ func TestKeycloakClientReconciler_Test_ClientScope_DifferenceIntersection(t *tes
 	assert.Equal(t, expectedDifference, difference)
 	assert.Equal(t, expectedIntersection, intersection)
 }
+
+func TestPodLabels_When_EnvVars_Then_FullListOfLabels(t *testing.T) {
+	cr := v1alpha1.Keycloak{
+		Spec: v1alpha1.KeycloakSpec{
+			KeycloakDeploymentSpec: v1alpha1.KeycloakDeploymentSpec{
+				PodLabels: map[string]string{
+					"LabelToTest":       "thisisthelabelvalue",
+					"SecondLabelToTest": "anotherthisisthelabelvalue",
+				},
+			},
+		}}
+
+	PodLabels = map[string]string{
+		"FirstLabel":  "firstLabelValue",
+		"SecondLabel": "secondLabelValue",
+	}
+
+	labels := map[string]string{
+		"app":       ApplicationName,
+		"component": KeycloakDeploymentComponent,
+	}
+	totalLabels := AddPodLabels(&cr, labels)
+	assert.Equal(t, 6, len(totalLabels))
```

Comment:
I realized we cannot hardcode this as the "constant" with default labels will be different in prod. We need to reflect the constant it this test.

Reply:
You mean in "keycloak-operator-prod" ? You suggest to extract that 6 into a constant that will be modified in the "keycloak-operator-prod" ? @vmuzikar

Reply:
Yes. In prod it might be 6 or 7 or 8 or ... theoretically any number of labels. It depends on how we modify the [`PodLabels`](https://github.com/keycloak/keycloak-operator/blob/ccf70429d9cff8e4e6036dd0163564a13a5d0c32/pkg/model/constants.go#L59) constant in prod.

Reply:
Ahhhh but take into account the test is NOT using what could be defined in the constants.go.
Few lines above the test is initializing the map : 
```
	PodLabels = map[string]string{
		"FirstLabel":  "firstLabelValue",
		"SecondLabel": "secondLabelValue",
	}

	labels := map[string]string{
		"app":       ApplicationName,
		"component": KeycloakDeploymentComponent,
	}
	totalLabels := AddPodLabels(&cr, labels)
	assert.Equal(t, 6, len(totalLabels))
```

Reply:
The test is using `AddPodLabels` which is using the constant:
https://github.com/keycloak/keycloak-operator/blob/ccf70429d9cff8e4e6036dd0163564a13a5d0c32/pkg/model/util.go#L224-L227

Reply:
Yes and AddPodLabels is using PodLabels, if you a closer look :
1. if it is not using the one few lines above the test won't pass
2. the PodLabels assignation few lines above is using the Global var

So the AddPodLabels is using the global var PodLabels that is initialized in the same test.

Reply:
Ah, sorry, you're right. I somehow managed to miss that you are assigning a new value to the `PodLabels` "constant" a few lines above.